### PR TITLE
[RTK] Migrate User component to rtk

### PR DESCRIPTION
**Pull request by dottharun**
**Notes for Reviewers**

This PR is related to #9985

- [x] replaced dataFetch calls with rtk-query calls
- [x] added error logs for rtk-query hooks
- [x] removed useEffects with useState logic when needed

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/User.js
Comment (sudhanshutech) [inline ui/components/User.js:80]: provide a proper and meaningful error message to user.
Use error property of rtk to get the actual error message.
```suggestion
        console.error('Error while fetching token ', error);
```

File: ui/components/User.js
Comment (sudhanshutech) [inline ui/components/User.js:92]: same here

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

**Comment by dottharun** [comment]
@sudhanshutech made the commit with changes, this should be enough i think

File: ui/components/User.js
Comment (leecalcote) [reply ui/components/User.js]: @sudhanshutech, in this case, should we be letting the user know of this error or no?

**Comment by saurabh100ni** [comment]
Hi @dottharun 
Thanks for your PR! I encourage you to share this in the Meshery Development Meeting tomorrow at 8:30 PM IST. Share your screen, and walk us through it. It's cool to take and share feedback. Check the [Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit) here to include this PR.

You're welcome at the meeting

Check to [community calendar](https://layer5.io/community/calendar) for more details of meetings at Layers.

File: ui/components/User.js
Comment (sudhanshutech) [reply ui/components/User.js]: yes it is good to show user a error here because from user side also there can be reason of error while fetching the token and having issues in logging. But we must use more exact info of error which comes from rtk itself

**Comment by sudhanshutech** [comment]
@dottharun you can take reference form `connection/index.js` in fetching environment , how error prop is used

**Comment by sudhanshutech** [comment]
@dottharun have you updated the spreadsheet linked to the issue

**Comment by sudhanshutech** [comment]
@dottharun have you updated the spreadsheet linked to the issue.
Otherwise look good please use the error prop from rtk. tagging @aabidsofi19 @Yashsharma1911 they can merge when you done

**Comment by dottharun** [comment]
@sudhanshutech Sorry, I'm unable to locate the spreadsheet. Any suggestions on where it might be?

File: ui/components/User.js
Comment (aabidsofi19) [inline ui/components/User.js:83]: dont use the effects please , that removes the benefit of using rtk

File: ui/components/User.js
Comment (aabidsofi19) [inline ui/components/User.js:100]: same here

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

**Comment by dottharun** [comment]
@aabidsofi19 changes are done and waiting for review

File: ui/components/User.js
Comment (aabidsofi19) [inline ui/components/User.js:88]: Can you confirm why the parent component uses this ,  it might make sense to use a direct rtk call there also

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/components/User.js
Comment (betelgeuse-code) [inline ui/components/User.js:95]: Hey @dottharun we can use `notify` hook here so that this issue gets displayed as toaster in UI, this issue needs user attention.

You can find implementation of that in Meshery codebase

Comment (betelgeuse-code) [reply ui/components/User.js:95]: Maybe in new iteration PR you can add that

**Review by betelgeuse-code — APPROVED** [review]

File: ui/components/User.js
Comment (betelgeuse-code) [reply ui/components/User.js]: Ohh Lee already mentioned my concern here @dottharun, ground here is that it is always a good idea to let user know the state of UI, in this by showing this error in toaster for which you can use `notify()` hook, however, if error if not blocking UI and can be avoided should go on console, I hope this clarifies the ground here

File: ui/components/User.js
Comment (betelgeuse-code) [reply ui/components/User.js:95]: https://github.com/meshery/meshery/pull/10270#discussion_r1505087614

Comment (betelgeuse-code) [reply ui/components/User.js:95]: https://github.com/meshery/meshery/pull/10270#discussion_r1505087614

Comment (dottharun) [reply ui/components/User.js:95]: @sudhanshutech do we want to use Notification center here?

**Comment by sudhanshutech** [comment]
@dottharun are you there? What's the status here

**Comment by dottharun** [comment]
> @dottharun are you there? What's the status here

Nothing more to do i guess

File: ui/components/User.js
Comment (Yashsharma1911) [inline ui/components/User.js:223]: This PR LGTM now, have  you made sure that you updated component name here in export from "User" to "UserProvider" you also updated same on all the places where this component is used if it is?

**Review by Yashsharma1911 — DISMISSED** [review]

File: ui/components/User.js
Comment (Yashsharma1911) [reply ui/components/User.js:223]: This PR is on hold for long let's make sure we merge this please sounds off on slack if you made sure of this test case

Comment (Yashsharma1911) [reply ui/components/User.js:223]: @dottharun other then this were you working asynchronously on other issues also?

File: ui/components/User.js
Comment (Yashsharma1911) [reply ui/components/User.js:95]: Yes please we want to use Notification center here, it always important to UI state with user, we can ignore those errors which doesn't block or affect UI, otherwise it's always good to show error to user

File: ui/components/User.js
Comment (dottharun) [reply ui/components/User.js:223]: This is a default export and User component is only used in Header.js. so I don't think we need to change anything else
- Also no other issues atm
Thanks

Comment (Yashsharma1911) [reply ui/components/User.js:223]: Thankyou

Comment (Yashsharma1911) [reply ui/components/User.js:223]: I wonder if you might misinterpreted my question here, I was checking code "UserProvider" component is not being used anywhere in codebase so why we even have the component? also I checked header.js in that is uses <User /> component from this file but "User" is not exported instead "UserProvider"?

Checkout header.js - https://github.com/dottharun/meshery/blob/e15ffc079935ececd0692706154d066b00b11e30/ui/components/Header.js#L15

Comment (dottharun) [reply ui/components/User.js:223]: Thanks for noting 
- you can check out nearby the end of the section [here](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#description), since it's a "default export" in can be renamed during import without any issue
- This pattern is used at other places too like [here](https://github.com/meshery/meshery/blob/e51ecf569e50c83fcbff0f8e5574691a98d044b6/ui/components/MesherySettings.js#L21)

**Review by Yashsharma1911 — APPROVED** [review]
LGTM

**Comment by Yashsharma1911** [comment]
@dottharun there are few other PRs raised by other contributors on RTK migration, maybe you would like to help by reviewing them?

**Comment by Yashsharma1911** [comment]
Please show this update in todays Meshery Dev Meeting if you didn't already

**Pull request merged**
